pkg/provisioner/ironic/ironic.go

```diff
@@ -354,6 +354,42 @@ func (p *ironicProvisioner) findExistingHost() (ironicNode *nodes.Node, err erro
 
 }
 
+func (p *ironicProvisioner) updateAutomatedClean(ironicNode *nodes.Node) (err error) {
+	var updates nodes.UpdateOpts
+	var value bool
+
+	p.log.Info("setting cleaning mode",
+		"ID", ironicNode.UUID,
+		"mode", p.host.Spec.AutomatedCleaningMode)
+
+	p.publisher("CleaningMode", string(p.host.Spec.AutomatedCleaningMode))
+
+	value = p.host.Spec.AutomatedCleaningMode != metal3v1alpha1.CleaningModeDisabled
+
+	updates = nodes.UpdateOpts{
+		nodes.UpdateOperation{
+			Op:    nodes.ReplaceOp,
+			Path:  "/automated_clean",
+			Value: value,
+		},
+	}
+
+	_, err = nodes.Update(p.client, ironicNode.UUID, updates).Extract()
```

Comment:
Wouldn't be better to request an update only _if_ required? Otherwise the call will be raised everytime (and the current changes on the unit tests are showing that)

Reply:
Knowing whether it's required can be tricky. The default value is `nil`, which means "use the default from ironic.conf". So you'll have to assume what's in ironic.conf.

Reply:
> Wouldn't be better to request an update only _if_ required? Otherwise the call will be raised everytime (and the current changes on the unit tests are showing that)

It would be, but `ironicNode.AutomatedClean` has nil by default as @dtantsur pointed out. As such, for example adding this:
```
if p.host.Spec.AutomatedCleaningMode == metal3v1alpha1.CleaningModeDisabled && *ironicNode.AutomatedClean != false || p.host.Spec.AutomatedCleaningMode == metal3v1alpha1.CleaningModeMetadata && *ironicNode.AutomatedClean == false {
    // do the update here
}
```
will cause runtime panic issue as we are using `*ironicNode.AutomatedClean`.
```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1484c44]
```
 We had the [slack chat](https://kubernetes.slack.com/archives/CHD49TLE7/p1615195502083900) about this with @dtantsur, and thought the way forward would be to always update it.

Reply:
Of course dereferencing a pointer without a nil check it's not feasible. And I don't think it's a blocker, considering that the call it's performed just in the provisioning/deprovisioning, but it would be nice to have a way to use that flag.

>The default value is nil, which means "use the default from ironic.conf". So you'll have to assume what's in ironic.conf.
Any way to retrieve the global conf in that case?

Reply:
There is no way to retrieve the global conf. What we could do is to avoid updates after the flag has been updated once (and its value is not `nil`).